### PR TITLE
Transaction logs API endpoint

### LINKED ISSUE #223: [Feature] - Add logs from transaction and scripts
### Issue To Be Solved
Since logs calls are not preserved on blockchain they are excluded from Access API, but they are crucial for debugging, testing and development process.

### Suggest A Solution
- Call for transaction result shall have `Logs` field
- Script execution result shall have `Logs` field 

### Context
Testing library is trying to extract and parse log statement from `stdout` stream, which is subpar and prone to errors.

### PR DESCRIPTION
**Pull request by bluesign**
Closes #223 

## Description

Partially closes #223. Added logs for transactions, but scripts are a bit mess there, see the discussion ( scripts always have the same ID for same script ) 

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: server/utils/emulator.go [hunk 1]
```diff
@@ -271,3 +273,34 @@ func (m EmulatorAPIServer) ResetCodeCoverage(w http.ResponseWriter, r *http.Requ
...
 }
+
+func (m EmulatorAPIServer) Logs(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	identifier, err := flowgo.HexStringToIdentifier(id)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (peterargue) [inline server/utils/emulator.go:284]: input is invalid
```suggestion
		w.WriteHeader(http.StatusBadRequest)
```

File: server/utils/emulator.go [hunk 2]
```diff
@@ -271,3 +273,34 @@ func (m EmulatorAPIServer) ResetCodeCoverage(w http.ResponseWriter, r *http.Requ
...
+			_ = json.NewEncoder(w).Encode([]string{})
+		} else {
+			_ = json.NewEncoder(w).Encode(logs)
+		}
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
+
```
Comment (peterargue) [inline server/utils/emulator.go:305]:
```suggestion
	if len(logs) == 0 {
		err = json.NewEncoder(w).Encode([]string{})
	} else {
		err = json.NewEncoder(w).Encode(logs)
	}

	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		return
	}
```

**Review by peterargue — APPROVED** [review]
added a couple small comments, but otherwise looks good

File: server/utils/emulator.go [hunk 3]
```diff
@@ -271,3 +273,34 @@ func (m EmulatorAPIServer) ResetCodeCoverage(w http.ResponseWriter, r *http.Requ
 	m.emulator.ResetCoverageReport()
 	w.WriteHeader(http.StatusOK)
 }
+
+func (m EmulatorAPIServer) Logs(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	vars := mux.Vars(r)
+	id := vars["id"]
```
Comment (turbolent) [inline server/utils/emulator.go:280]: Maybe check if the ID was provided and return a (ideally helpful) error (`StatusBadRequest`) if it is missing

**Review by turbolent — APPROVED** [review]
Looks good after @peterargue's improvements are in

**Review by janezpodhostnik — APPROVED** [review]

File: server/utils/emulator.go (hunk 3 above)
Comment (bluesign) [reply server/utils/emulator.go:280]: fixed at identifier check

**Pull request merged**